sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2603,12 +2603,27 @@ class Dataset[T] private[sql](
    * current upstream partitions will be executed in parallel (per whatever
    * the current partitioning is).
    *
+   * A [[PartitionCoalescer]] can also be supplied allowing the behavior of the partitioning to be
```

Comment:
Sounds this trait is unable to be generated as is in Java. Simply wrapping `` `...` `` would be fine.